### ISSUE TITLE
documentation for gems broken since 0.9.6

### ISSUE THREAD

**Issue by jsirex** [lsegal/yard#1058]
May be I'm doing something wrong due to new release, but:

```bash
╰─$ yard server --gems
[info]: No yardoc db found in .yardoc, parsing source before starting server...
Files:           0
Modules:         0 (    0 undocumented)
Classes:         0 (    0 undocumented)
Constants:       0 (    0 undocumented)
Attributes:      0 (    0 undocumented)
Methods:         0 (    0 undocumented)
 100.00% documented
>> YARD 0.9.7 documentation server at http://localhost:8808
[2017-01-13 12:52:35] INFO  WEBrick 1.3.1
[2017-01-13 12:52:35] INFO  ruby 2.3.1 (2016-04-26) [x86_64-linux]
[2017-01-13 12:52:35] INFO  WEBrick::HTTPServer#start: pid=14313 port=8808
```

In gems mode it tries to document my home directory as a project. So as a result, no docs at all.
Same behavior with 0.9.6. But it works as expected with 0.9.5.

# OS Info
Debian GNU/Linux (Testing) + RVM

```
╰─$ rvm info

ruby-2.3.1:

  system:
    uname:       "Linux sirex-notebook 4.8.0-2-amd64 #1 SMP Debian 4.8.11-1 (2016-12-02) x86_64 GNU/Linux"
    system:      "debian/stretch_sid/x86_64"
    bash:        "/usr/bin/bash => GNU bash, version 4.4.5(1)-release (x86_64-pc-linux-gnu)"
    zsh:         "/usr/bin/zsh => zsh 5.3 (x86_64-debian-linux-gnu)"

  rvm:
    version:      "rvm 1.27.0 (latest) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]"
    updated:      "2 months 14 days 21 hours 21 minutes 35 seconds ago"
    path:         "/home/sirex/.rvm"

  ruby:
    interpreter:  "ruby"
    version:      "2.3.1p112"
    date:         "2016-04-26"
    platform:     "x86_64-linux"
    patchlevel:   "2016-04-26 revision 54768"
    full_version: "ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-linux]"

  homes:
    gem:          "/home/sirex/.rvm/gems/ruby-2.3.1"
    ruby:         "/home/sirex/.rvm/rubies/ruby-2.3.1"

  binaries:
    ruby:         "/home/sirex/.rvm/rubies/ruby-2.3.1/bin/ruby"
    irb:          "/home/sirex/.rvm/rubies/ruby-2.3.1/bin/irb"
    gem:          "/home/sirex/.rvm/rubies/ruby-2.3.1/bin/gem"
    rake:         "/home/sirex/.rvm/gems/ruby-2.3.1/bin/rake"

  environment:
    PATH:         "/home/sirex/.rvm/gems/ruby-2.3.1/bin:/home/sirex/.rvm/gems/ruby-2.3.1@global/bin:/home/sirex/.rvm/rubies/ruby-2.3.1/bin:/usr/local/bin:/usr/bin:/bin:/usr/games:/home/sirex/.rvm/bin:/home/sirex/.rvm/bin"
    GEM_HOME:     "/home/sirex/.rvm/gems/ruby-2.3.1"
    GEM_PATH:     "/home/sirex/.rvm/gems/ruby-2.3.1:/home/sirex/.rvm/gems/ruby-2.3.1@global"
    MY_RUBY_HOME: "/home/sirex/.rvm/rubies/ruby-2.3.1"
    IRBRC:        "/home/sirex/.rvm/rubies/ruby-2.3.1/.irbrc"
    RUBYOPT:      ""
    gemset:       ""

```
Gems:
```
╰─$ gem list

*** LOCAL GEMS ***

addressable (2.5.0, 2.4.0)
artifactory (2.5.1, 2.5.0)
ast (2.3.0)
backports (3.6.8)
berkshelf (5.3.0, 5.2.0, 5.1.0)
berkshelf-api-client (3.0.0)
bigdecimal (1.2.8)
bson (4.2.1, 4.1.1)
buff-config (2.0.0)
buff-extensions (2.0.0)
buff-ignore (1.2.0)
buff-ruby_engine (1.0.0)
buff-shell_out (1.1.0)
builder (3.2.2)
bundler (1.13.6)
bundler-unload (1.0.2)
byebug (9.0.6)
celluloid (0.16.0)
celluloid-io (0.16.2)
chef (12.17.44, 12.16.42, 12.15.19, 12.14.89)
chef-api (0.7.0)
chef-config (12.17.44, 12.16.42, 12.15.19, 12.14.89)
chef-zero (5.1.1, 5.1.0)
chefspec (5.3.0)
ci_reporter (2.0.0)
ci_reporter_rspec (1.0.0)
cleanroom (1.0.0)
coderay (1.1.1)
cucumber-core (2.0.0)
did_you_mean (1.0.0)
diff-lcs (1.2.5)
domain_name (0.5.20161129, 0.5.20160826)
e3s-toolbelt (2.8.3, 2.8.1)
elasticsearch (1.1.0)
elasticsearch-api (1.1.0)
elasticsearch-transport (1.1.0)
epath (0.1.1)
erubis (2.7.0)
executable-hooks (1.3.2)
faraday (0.9.2)
fauxhai (3.10.0)
ffi (1.9.14)
ffi-yajl (2.3.0)
foodcritic (8.2.0, 8.1.0, 8.0.0)
formatador (0.2.5)
fuzzyurl (0.9.0)
gem-wrappers (1.2.7)
gherkin (4.0.0)
gitlab (3.7.0)
guard (2.14.0)
guard-bundler (2.1.0)
guard-compat (1.2.1)
guard-rspec (4.7.3)
guard-rubocop (1.2.0)
hashie (3.4.6)
highline (1.7.8)
hitimes (1.2.4)
http-cookie (1.0.3)
httparty (0.13.7)
httpclient (2.8.3, 2.8.2.4)
iniparse (1.4.2)
io-console (0.4.5)
ipaddress (0.8.3)
json (2.0.2, 1.8.3)
kitchen-docker (2.6.0)
kitchen-sync (2.1.1)
kitchen-vagrant (0.21.1, 0.21.0, 0.20.0)
knife-block (0.2.1)
knife-cookbook-doc (0.21.0, 0.20.0)
knife-solo_data_bag (2.0.0, 1.1.0)
knife-supermarket (0.3.0)
libyajl2 (1.2.0)
listen (3.1.5)
logify (0.2.0)
lumberjack (1.0.11, 1.0.10)
method_source (0.8.2)
mime-types (3.1, 2.99.3)
mime-types-data (3.2016.0521)
mini_portile2 (2.1.0)
minitar (0.5.4)
minitest (5.8.3)
mixlib-archive (0.2.0)
mixlib-authentication (1.4.1)
mixlib-cli (1.7.0)
mixlib-config (2.2.4)
mixlib-install (2.1.9, 2.1.8, 2.1.7, 2.1.5)
mixlib-log (1.7.1)
mixlib-shellout (2.2.7)
mixlib-versioning (1.1.0)
molinillo (0.5.5, 0.5.4, 0.5.3, 0.5.1)
mongo (2.4.1, 2.3.0)
multi_json (1.12.1)
multi_xml (0.5.5)
multipart-post (2.0.0)
nenv (0.3.0)
net-scp (1.2.1)
net-sftp (2.1.2)
net-ssh (3.2.0)
net-ssh-gateway (1.2.0)
net-ssh-multi (1.2.1)
net-telnet (0.1.1)
netrc (0.11.0)
nexus (1.2.3)
nio4r (2.0.0, 1.2.1)
noe (1.7.6)
nokogiri (1.7.0.1, 1.7.0, 1.6.8.1, 1.6.8)
notiffany (0.1.1)
octokit (4.6.2, 4.4.1, 4.3.0)
ohai (8.22.1, 8.22.0, 8.21.0, 8.20.0)
parser (2.3.3.1, 2.3.3.0, 2.3.1.4)
pkg-config (1.1.7)
plist (3.2.0)
polyglot (0.3.5)
power_assert (0.2.6)
powerpack (0.1.1)
proxifier (1.0.3)
pry (0.10.4)
pry-byebug (3.4.2)
psych (2.0.17)
public_suffix (2.0.5, 2.0.4)
quickl (0.4.3)
rack (2.0.1)
rainbow (2.2.1, 2.1.0)
rake (12.0.0, 11.3.0, 10.4.2)
rb-fsevent (0.9.8)
rb-inotify (0.9.7)
rdoc (5.0.0, 4.2.2, 4.2.1)
rest-client (1.8.0)
retryable (2.0.4)
ridley (5.1.0)
rspec (3.5.0)
rspec-core (3.5.4, 3.5.3)
rspec-expectations (3.5.0)
rspec-its (1.2.0)
rspec-mocks (3.5.0)
rspec-support (3.5.0)
rspec_junit_formatter (0.2.3)
rubocop (0.46.0, 0.45.0)
ruby-graphviz (1.2.2)
ruby-progressbar (1.8.1)
ruby_dep (1.5.0)
rubygems-bundler (1.4.4)
rufus-lru (1.1.0)
rvm (1.11.3.9)
safe_yaml (1.0.4)
sawyer (0.8.1, 0.7.0)
semverse (2.0.0)
serverspec (2.37.2, 2.36.1)
sfl (2.3, 2.2)
shellany (0.0.1)
slop (3.6.0)
solve (3.1.0, 3.0.1)
specinfra (2.66.3, 2.66.2, 2.66.0, 2.64.0, 2.63.1)
sshkit (1.11.5, 1.11.3)
stove (4.1.1)
syslog-logger (1.6.8)
systemu (2.6.5)
terminal-table (1.7.3)
test-kitchen (1.14.2, 1.14.0, 1.13.2)
test-unit (3.1.5, 2.5.5)
thor (0.19.4, 0.19.1)
timers (4.0.4)
treetop (1.6.8)
unf (0.1.4)
unf_ext (0.0.7.2)
unicode-display_width (1.1.2, 1.1.1)
uuidtools (2.1.5)
varia_model (0.6.0)
waitutil (0.2.1)
wlang (0.10.2)
wmi-lite (1.0.0)
yajl-ruby (1.3.0, 1.2.1)
yard (0.9.7, 0.9.6, 0.9.5)
```

Also, because I unable to test with latest (0.9.7), I have another issue, probably fixed:
I installed chefspec (5.3.0) gem. But unable to get to it documentation. Debug + Verbose shows me infinitive cycle of parsing.

**Comment by lsegal** [comment]
Thanks for reporting this! Fix should be coming soon.
